rpm/stackdriver-agent

```diff
@@ -118,57 +164,107 @@ start () {
     fi
 
     if ! check_config; then
-        echo >&2 "Not starting, configuration/credentials error."
-        return 1
+        log_failure_msg "not starting, configuration/credentials error."
+        return 3
     fi
 
-    if [ -r "$CONFIG" ]; then
-        LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/opt/stackdriver/collectd/lib64:/opt/stackdriver/collectd/lib" start_daemon "$DAEMON" -C "$CONFIG" -P "$pidfile"
-        RETVAL=$?
-        echo
-        [ $RETVAL -eq 0 ] && touch /var/lock/subsys/$prog
-    else
-        echo "Can't read agent configuration file: $CONFIG"
-        RETVAL=1
-    fi
-    return $RETVAL
+    LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/opt/stackdriver/collectd/lib64:/opt/stackdriver/collectd/lib" \
+        start_daemon -p "$_PIDFILE" "$DAEMON" -C "$CONFIG" -P "$_PIDFILE" \
+        || return 2
+    return 0
 }
 
-stop () {
-    echo -n $"Stopping $prog: "
-    killproc $prog
-    RETVAL=$?
-    echo
+# return:
+#   0 if the daemon has been stopped
+#   1 if the daemon could not be stopped
+#   3 if the daemon was already stopped
+stop() {
+    killproc -p "$_PIDFILE" "$DAEMON"
+}
 
-    [ $RETVAL -eq 0 ] && rm -f /var/lock/subsys/$prog
+# return:
+#   0 if the daemon has been restarted
+#   1 if the daemon could not be stopped
+#   2 if the daemon could not be started
+#   3 if the daemon was not supposed to be restarted
+restart() {
+    if ! check_config; then
+        log_failure_msg "not restarting, configuration/credentials error"
+        return 3
```

Comment:
Similar here: `3 if the daemon was not supposed to be restarted` sounds like it's working as expected. Yet in fact we are failing to restart the agent. Should we return some other code or rephrase the exit code `3`'s description?

Reply:
See above.